src/wizards/binary/buildBinary.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2021 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { env, Task, tasks, Uri, window, workspace, WorkspaceFolder } from "vscode";
+import { BuildSupport } from "../../buildSupport/BuildSupport";
+import { ProjectLabelInfo } from "../../definitions/ProjectLabelInfo";
+import { TaskCreator } from "../debugging/TaskCreator";
+import { getQuarkusProject } from "../getQuarkusProject";
+
+const NATIVE_IMAGE_DOCS_URI: Uri = Uri.parse('https://quarkus.io/guides/building-native-image');
+
+const RUN_ANYWAYS = 'Run anyways';
+const DO_NOT_RUN = 'Don\'t run';
+const LEARN_MORE = 'Learn more...';
+
+/**
+ * Selects a quarkus project, then starts building a binary for it
+ *
+ * Prompts the user if GraalVM isn't detected
+ *
+ * @returns when the binary build task has been started
+ */
+export async function buildBinary(): Promise<void> {
+  const projectLabelInfo: ProjectLabelInfo = await getQuarkusProject();
+  if (!projectLabelInfo) {
+    return;
+  }
+  if (!hasGraalVmHome()) {
+    const result = await window.showInformationMessage(
+      'The environment variable GRAALVM_HOME is not set, '
+      + 'meaning the binary build will likely fail. '
+      + 'Attempt to generate a binary anyways?',
+      DO_NOT_RUN, RUN_ANYWAYS, LEARN_MORE);
+    if (result === LEARN_MORE) {
+      env.openExternal(NATIVE_IMAGE_DOCS_URI);
+    }
+    if (result !== RUN_ANYWAYS) {
+      return;
+    }
+  }
+  try {
+    await buildBinaryFromProject(projectLabelInfo);
+  } catch (e) {
+    window.showErrorMessage(`${e}`);
+  }
+}
+
+/**
+ * Returns true if GRAALVM_HOME is set and false otherwise
+ *
+ * @returns true if GRAALVM_HOME is set and false otherwise
+ */
+function hasGraalVmHome(): boolean {
+  return !!process.env.GRAALVM_HOME.trim();
```

Comment:
Running without GRAALVM_HOME set : `Exception has occurred: TypeError: Cannot read property 'trim' of undefined`